homeassistant/helpers/config_validation.py

```diff
@@ -423,12 +423,16 @@ def remove_falsy(value: List[T]) -> List[T]:
     return [v for v in value if v]
 
 
-def service(value: Any) -> str:
+def service(value: Any) -> Union[str, template_helper.Template]:
```

Comment:
We don't know where else this validator is used, so probably better to create a new validator that can return a string or template.

Reply:
I could not find any references, nevertheless, yeah that can be done.
The service schema could be used elsewhere as well, so should we recreate that as well?

Reply:
I would say no, because that schema is meant to be passed to our service helper.

Reply:
I doubt anyone really uses the `service` helper, but maybe better safe than sorry?

Reply:
```python
try:
  return service(val)
except vol.Invalid:
    if template_helper.is_template_string(value):
        return template(value)
    raise
```

Reply:
Alternative implemented in b71a32a